extension/app/src/components/conversation/UserMessage.tsx

```diff
@@ -50,6 +77,86 @@ export function UserMessage({
     []
   );
 
+  const configurationIds = message.mentions.map(
+    (mention) => mention.configurationId
+  );
+
+  useEffect(() => {
+    const loadConfigurations = async () => {
+      const saved = await getSavedConfigurations();
+      if (saved.length !== savedConfigurations.length) {
+        setSavedConfigurations(saved);
+      }
+    };
+    void loadConfigurations();
+  }, []);
+
+  useEffect(() => {
+    const getId = async () => {
+      const hash = await hashBase64(`${configurationIds}-${message.content}`);
+      const id = `ask_dust_${hash}`;
+      setSavedConfigurationId(id);
+    };
+    void getId();
+  }, []);
+
+  const enabled = !!savedConfigurations.find(
+    (c) => c.id === savedConfigurationId
+  );
+
+  const buttons = isFirstMessage
+    ? [
+        <Button
+          variant="outline"
+          size="xs"
+          icon={enabled ? HeartIcon : HeartStrokeIcon}
+          label={enabled ? "Remove from context menu" : "Save to context menu"}
```

Comment:
I don't think "context menu" is easily understandable, I suggest "Save as Quick Action" / "Remove from Quick Actions" 🙏🏻